var/spack/repos/builtin/packages/charliecloud/package.py

```diff
@@ -22,6 +24,13 @@ class Charliecloud(MakefilePackage):
     version('0.9.0', sha256='7e74cb16e31fd9d502198f7509bab14d1049ec68ba90b15e277e76f805db9458')
     version('0.2.4', 'b112de661c2c360174b42c99022c1967')
 
+    def url_for_version(self, version):
+        if version >= Version('0.9.8'):
+            url = "https://github.com/hpc/charliecloud/releases/download/v0.9.10/charliecloud-0.9.10.tar.gz"
+        else:
+            url = "https://github.com/hpc/charliecloud/archive/v0.2.4.tar.gz"
+        return url.format(version)
```

Comment:
This isn't going to work. You need format strings inside of this function. When I said to get rid of the format strings, I meant above in the package-level `url`.